spec/factories/products.rb

```diff
@@ -3,5 +3,6 @@
     name { "MyString" }
     description { "MyText" }
     price { 1 }
+    category { "myString" }
```

Comment:
So category can not be a string. try just 
```
category
```
FactoryBot will create a category object.